transforms/parse5Utils.js

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import * as parse5Module from 'parse5'
+
+const parse5 = parse5Module.default ?? parse5Module
+
+export const walkParse5Ast = (node, cb, ctx) => {
+  if (cb) {
+    cb(node, ctx)
+  }
+
+  const children = node.tagName === 'template' ? node.content.childNodes : node.childNodes
+  // Not all parse5 AST nodes have childNodes
+  for (const child of children ?? []) {
+    walkParse5Ast(child, cb, ctx)
+  }
+}
+
+export const serializeParse5Ast = (ast) =>
+// replace the quotes around `foo="{bar}"` which parse5 adds, but the LWC compiler doesn't like
+  parse5.serialize(ast).replace(/(\w+)="{([\w.]+)+}"/g, '$1={$2}')
+
+export const getAttr = (node, name) => {
+  const attr = node.attrs ? node.attrs.find((attr) => attr.name === name) : null
+  return attr && attr.value
+}
+
+export const setAttr = (node, name, value) => {
+  const attr = node.attrs.find((attr) => attr.name === name)
+  if (attr) {
+    attr.value = value
+  } else {
+    node.attrs.push({ name, value })
+  }
+}
```

Comment:
I moved these out of the `modifyTemplateHtml.js` file because I thought it might be helpful for other mods in the future.

Currently, I don't use them on the other PRs but might in the future.